File: frontend/src/api/watchlist.js
import axios from 'axios'

const api = axios.create({
  baseURL: `${import.meta.env.VITE_API_BASE || '/api/v1'}/watchlist`,
  timeout: 30000
})

export default {
  list() {
    return api.get('').then((r) => r.data)
  },
  add(symbol, groupId = 'default', name = '', note = '') {
    return api.post('/items', { symbol, group_id: groupId, name, note }).then((r) => r.data)
  },
  remove(symbol, groupId = 'default') {
    return api.delete(`/items/${symbol}`, { params: { group_id: groupId } }).then((r) => r.data)
  },
  createGroup(name) {
    return api.post('/groups', { name }).then((r) => r.data)
  }
}

File: backend/app/api/watchlist.py
from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import Optional

from app.services.watchlist import get_watchlist_service

router = APIRouter()


class WatchlistAdd(BaseModel):
    symbol: str
    group_id: str = "default"
    name: str = ""
    note: str = ""


class WatchlistGroupCreate(BaseModel):
    name: str


@router.get("")
async def list_watchlist():
    try:
        return {"groups": get_watchlist_service().list_groups()}
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"自选股读取失败: {e}") from e


@router.post("/items")
async def add_watchlist_item(body: WatchlistAdd):
    try:
        item = get_watchlist_service().add_item(
            body.symbol, body.group_id, body.name, body.note
        )
        return item
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))


@router.delete("/items/{symbol}")
async def remove_watchlist_item(symbol: str, group_id: str = "default"):
    if not get_watchlist_service().remove_item(symbol, group_id):
        raise HTTPException(status_code=404, detail="标的不在自选中")
    return {"ok": True}


@router.post("/groups")
async def create_group(body: WatchlistGroupCreate):
    return get_watchlist_service().create_group(body.name)
